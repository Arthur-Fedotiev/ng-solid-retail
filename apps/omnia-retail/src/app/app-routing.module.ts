import { NgModule } from '@angular/core';
import { Route, RouterModule } from '@angular/router';
import { LayoutFeatureComponent } from '@omnia/layout/feature';

const appRoutes: Route[] = [
  {
    path: '',
    component: LayoutFeatureComponent,
    children: [
      {
        path: 'products',
        loadChildren: () =>
          import('@omnia/products/shell').then((m) => m.ProductsShellModule),
      },
    ],
  },
];

@NgModule({
  imports: [
    RouterModule.forRoot(appRoutes, {
      enableTracing: false,
    }),
  ],
  exports: [RouterModule],
})
export class AppRoutingModule {}
