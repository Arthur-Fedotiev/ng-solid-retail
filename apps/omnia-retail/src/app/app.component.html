<omnia-nx-welcome></omnia-nx-welcome>
