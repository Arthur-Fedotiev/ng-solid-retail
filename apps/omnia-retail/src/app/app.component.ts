import { ChangeDetectionStrategy, Component } from '@angular/core';

@Component({
  selector: 'omnia-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AppComponent {}
