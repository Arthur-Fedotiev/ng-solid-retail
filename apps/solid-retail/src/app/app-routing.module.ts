import { NgModule } from '@angular/core';
import { Route, RouterModule } from '@angular/router';
import { LayoutFeatureComponent } from '@sr/layout/feature';

const appRoutes: Route[] = [
  {
    path: '',
    component: LayoutFeatureComponent,
    children: [
      {
        path: 'products',
        loadChildren: () =>
          import('@sr/products/shell').then((m) => m.ProductsShellModule),
      },
      {
        path: '**',
        redirectTo: 'products',
      },
    ],
  },
];

@NgModule({
  imports: [
    RouterModule.forRoot(appRoutes, {
      enableTracing: false,
    }),
  ],
  exports: [RouterModule],
})
export class AppRoutingModule {}
