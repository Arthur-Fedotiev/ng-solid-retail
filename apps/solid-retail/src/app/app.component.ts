import { ChangeDetectionStrategy, Component } from '@angular/core';
import { RouterOutlet } from '@angular/router';

@Component({
  selector: 'sr-root',
  styleUrls: ['./app.component.scss'],
  standalone: true,
  imports: [RouterOutlet],
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: `<router-outlet></router-outlet>`,
})
export class AppComponent {}
