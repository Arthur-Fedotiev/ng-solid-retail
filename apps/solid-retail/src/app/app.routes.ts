import { Route } from '@angular/router';
import { LayoutFeatureComponent } from '@sr/layout/feature';

export const appRoutes: Route[] = [
  {
    path: '',
    component: LayoutFeatureComponent,
    children: [
      {
        path: 'products',
        loadChildren: () =>
          import('@sr/products/shell').then((m) => m.PRODUCT_ROUTES),
      },
      {
        path: '**',
        redirectTo: 'products',
      },
    ],
  },
];
