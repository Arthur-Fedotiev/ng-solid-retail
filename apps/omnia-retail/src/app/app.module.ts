import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { LayoutFeatureModule } from '@omnia/layout/feature';
import { AppRoutingModule } from './app-routing.module';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

import { AppComponent } from './app.component';
import { HttpClientModule } from '@angular/common/http';

@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    HttpClientModule,
    LayoutFeatureModule,
    AppRoutingModule,
  ],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
