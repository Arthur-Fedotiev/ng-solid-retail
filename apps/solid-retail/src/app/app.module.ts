import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { LayoutFeatureModule } from '@sr/layout/feature';
import { AppRoutingModule } from './app-routing.module';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

import { AppComponent } from './app.component';
import { HttpClientModule } from '@angular/common/http';

import { AngularFireModule } from '@angular/fire/compat';
import { AngularFirestoreModule } from '@angular/fire/compat/firestore';
import { environment } from '@sr/shared/environments';

@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    HttpClientModule,
    LayoutFeatureModule,
    AppRoutingModule,
    AngularFireModule.initializeApp(environment.firebaseConfig),
    AngularFirestoreModule,
  ],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
