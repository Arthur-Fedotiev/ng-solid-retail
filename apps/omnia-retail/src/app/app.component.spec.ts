import { ComponentFixture, TestBed } from '@angular/core/testing';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [AppRoutingModule],
      declarations: [AppComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;

    fixture.detectChanges();
  });

  it('should create the app', () => {
    expect(component).toBeTruthy();
  });

  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });
});
