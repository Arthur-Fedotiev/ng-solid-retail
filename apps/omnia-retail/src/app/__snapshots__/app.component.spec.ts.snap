// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render 1`] = `
<omnia-root
  __ngContext__={[Function Number]}
>
  <router-outlet />
</omnia-root>
`;
